### PR TITLE
:shell: Add pharmacy opening times

### PR DESCRIPTION
**Pull request by neilbmclaughlin**

### CODE REVIEW

**Comment by neilbmclaughlin** [comment]
Rebased.

Currently just displays Monday opening times in a very ugly fashion.

File: app/lib/openingTimesParser.js [hunk 1]
```diff
@@ -14,38 +14,40 @@ const parseGpOpeningTimesFromSyndicationXml = (openingTimesType, xml) => {
   const gpOpeningTimes = {};
```
Comment (st3v3nhunt) [inline app/lib/openingTimesParser.js]: Rename variable to `openingTimes`?

File: app/lib/openingTimesParser.js [hunk 2]
```diff
@@ -14,38 +14,40 @@ const parseGpOpeningTimesFromSyndicationXml = (openingTimesType, xml) => {
...
         data: openingTimes,
         locals: {
           isType: (item) => item.$.sessionType === openingTimesType,
         },
       }).value[0];
 
-      openingTimesForType.daysOfWeek.dayOfWeek.forEach((item) => {
-        const dayName = item.dayName.toLowerCase();
+      openingTimesForType.daysOfWeek[0].dayOfWeek.forEach((item) => {
+        const dayName = item.dayName[0].toLowerCase();
```
Comment (st3v3nhunt) [inline app/lib/openingTimesParser.js]: Is this going to be where the logic to check the current day goes?

File: app/middleware/gp.js [hunk 3]
```diff
@@ -75,36 +118,23 @@ function getDetails(req, res, next) {
 }
 
 function getOpeningTimes(req, res, next) {
-  http.get(req.gpDetails.overviewLink, (response) => {
-    let syndicationXml = '';
-    response.on('data', (chunk) => {
-      syndicationXml += chunk;
-    });
+  assert.ok(validUrl.isUri(req.gpDetails.overviewLink),
```
Comment (st3v3nhunt) [inline app/middleware/gp.js]: `req.gpDetails` should this be named more generically?

**Comment by st3v3nhunt** [comment]
The name of the file `app/middleware/gp.js` is probably in need of a rename given it now houses non-gp functionality. There are a few variables in there that might also need to be renamed to something more generic

File: app/lib/openingTimesParser.js (hunk 2 above)
Comment (neilbmclaughlin) [inline app/lib/openingTimesParser.js]: Probably in the middleware

**Pull request merged**